### PR TITLE
Add __eq__ and __hash__ to NamedUser

### LINKED ISSUE #608: Implement __hash__() and __eq__() on NamedUser
I have a case where I'm processing the list of reviewers on a Pull Request. When the same person does two reviews the `NamedUser` return from the `PullRequest.get_reviews()`, while they have the same `login`, `id`, and `name`, don't have the same hash value. 

If you implement `__hash__()` the python docs strongly recommend implementing `__eq__()` based upon the same values. 

It would be nice to have the python build in functions be able to construct the unique list of reviewers.

### PR DESCRIPTION
**Pull request by namratachaudhary**
Resolves #608

### CODE REVIEW

File: github/NamedUser.py [hunk 1]
```diff
@@ -46,6 +46,14 @@ class NamedUser(github.GithubObject.CompletableGithubObject):
     def __repr__(self):
         return self.get__repr__({"login": self._login.value})
 
+    def __hash__(self):
+        return hash((self.id, self.login))
+
+    def __eq__(self, other):
+        return (isinstance(other, type(self))
```
Comment (sfdye) [inline github/NamedUser.py]: Thanks for the change. Even the most outer pair of brackets are not need.
```python
return isinstance(other, type(self)) and self.login == other.login and self.id == other.id)
```

**Comment by sfdye** [comment]
@namratachaudhary LGTM, I will merge once you remove the brackets.

File: github/NamedUser.py (hunk 1 above)
Comment (namratachaudhary) [reply github/NamedUser.py]: Ah! The linter was throwing an error, so I hesitated. Will make the change nevertheless

Comment (sfdye) [reply github/NamedUser.py]: What did the linter say?

**Comment by sfdye** [comment]
@namratachaudhary Merged. Thank you for your contribution!

**Pull request merged**